fastapi_opa/auth/auth_saml.py

```diff
@@ -29,15 +33,60 @@ async def authenticate(
         auth = await self.init_saml_auth(request_args)
 
         if "acs" in request.query_params:
+            logger.debug(datetime.utcnow(), '--acs--')
             return await self.assertion_consumer_service(auth, request_args)
         # potentially extend with logout here
+        elif 'sso' in request.query_params:
+            logger.debug(datetime.utcnow(), '--sso--')
+            return await self.single_sign_on(auth)
+            # TODO: check below code
+            # If AuthNRequest ID need to be stored in order to later validate it, do instead
+            # sso_built_url = auth.login()
+            # request.session['AuthNRequestID'] = auth.get_last_request_id()
+            # return redirect(sso_built_url)
+        elif 'sso2' in request.query_params:
+            logger.debug(datetime.utcnow(), '--sso2--')
+            return_to = '%sattrs/' % request.base_url
+            return RedirectResponse(auth.login(return_to))
+        elif 'slo' in request.query_params:
+            logger.debug(datetime.utcnow(), '--slo--')
+            return await self.single_log_out(auth)
+        # TODO: handle sls
+        # elif 'sls' in request.query_params:
+        #     logger.debug(datetime.utcnow(), '--sls--')
+        #     request_id = None
+        #     if 'LogoutRequestID' in request.query_params['post_data']:
+        #         request_id = req_args['post_data']['LogoutRequestID']
+        #     # TODO: not sure how to handle session here
+        #     dscb = lambda: request.session.flush()
+        #     url = auth.process_slo(request_id=request_id, delete_session_cb=dscb)
+        #     errors = auth.get_errors()
+        #     if len(errors) == 0:
+        #         if url is not None:
+        #             return RedirectResponse(url)
+        #         else:
+        #             success_slo = True
+        #     elif auth.get_settings().is_debug_active():
+        #         error_reason = auth.get_last_error_reason()
         return await self.single_sign_on(auth)
 
     async def init_saml_auth(self, request_args: Dict) -> OneLogin_Saml2_Auth:
         return OneLogin_Saml2_Auth(
             request_args, custom_base_path=self.custom_folder.as_posix()
         )
 
+    @staticmethod
+    async def single_log_out(auth: OneLogin_Saml2_Auth) -> RedirectResponse:
+
+        name_id = auth.get_nameid() or None
```

Comment:
I did not check all of the methods but I guess they are all returning `None` or `str`:
https://github.com/onelogin/python3-saml/blob/201ea4b32a8658a43494c2406ef6fc525bbdbecf/src/onelogin/saml2/auth.py#L252-L259

Therefore `or None` is not necessary there.